### ISSUE TITLE
ImageBERT: Cross-modal Pre-training with Large-scale Weak-supervised Image-Text Data

### ISSUE THREAD

**Issue by reo11** [reo11/papers100knock#2]
## 概要
- 文章に加えて画像に対してもマスクを行ない学習
- 画像に対しては、画像のラベルや特徴（画像の例では部位）を予測
- 最後に文章と画像が一致しているかの予測（NSPに相当？）

## 投稿年/会議
2020年/ArXiv
- https://arxiv.org/pdf/2001.07966.pdf

## 著者/所属機関
Di Qi, Lin Su, Jia Song, Edward Cui, Taroon Bharti, Arun Sacheti
- Bing Multimedia Team, **Microsoft**

## 参考リンク
- https://qrunch.net/@opqrstuvcut/entries/LgGT7PweQjkkz9kN?ref=qrunch
- https://github.com/arXivTimes/arXivTimes/issues/1562
- https://medium.com/syncedreview/microsoft-imagebert-cross-modal-pretraining-with-large-scale-image-text-data-90d0e9c3c97a

### 使用したデータセット
- Conceptual Captions (CC) Datasets
  - 300万画像
- SBU Captions
  - 100万画像
- Large-scale weAk-supervised Image-Text (LAIT)
  - ウェブからクロールし、ノイズとなるような投稿を除去したデータセットを作成
  - 平均13単語の注釈が付いたデータセット
  - 1000万画像
### アーキテクチャ
![image](https://user-images.githubusercontent.com/38539915/79503251-c8fb7480-806b-11ea-8fcb-20dbef4d69ad.png)
- Linguistic embedding
  - 普通のBERTと同じようにサブワードに分割
- Image embedding
  - ImageBERTに入れる前にFaster-RCNNでembedding化
  - この時、領域検出とラベルの検出が行われるのでそれらを使用
  - 領域情報はPosition Embeddingとして入力
  - 領域情報は画像内の相対位置や相対面積として入力される（解像度に影響されない）
  - p.5のe(i)の数式が図と整合していないように見えた（Position EmbeddingとImage EmbeddingでLinguistic Embeddingと同じ次元にしているのかと思ったが、4つ並列に扱っている）
- 共通するもの
  - Sequence Position Embedding
    - 画像には順序がないため、全部1
    - テキストに対しては2, 3, ..., n-1
  - Segment Embedding
     - 画像（A）とテキスト（B）別々のラベルを付ける

### Multi-stage Pretraining
![image](https://user-images.githubusercontent.com/38539915/79507226-504be680-8072-11ea-9d78-f619ffd20706.png)
- 学習に使用できるデータが少なく、webクロールしたデータ（LAIT）を使用しているため、複数のステージに分けて学習
- 1～k+1ステージ目はLAIT、k+2ステージ目はCC+SBUでpre-train

### Pre-training tasks
1. Masked Language Modeling (MLM)
  - BERTと同じMLMタスク
  - 15%にマスク処理
  - [MASK]に置き換え、ランダムトークン、変更なしの割合＝8:1:1
2. Masked Object Classification (MOC)
  - object tokenを15%の確率でマスク
  - マスクされたトークンを削除、そのままの割合9:1
3. Masked Region Feature Regression (MRFR)
  - 入力画像から生成される特徴量とImageBERTで出力する特徴量ベクトルの次元を合わせ、L2を最小化
4. Image-Text Matching (ITM)
  - 画像とテキストが一致しているか
  - 収集したデータの画像とテキストを適当にシャッフルしてnegative sampleを生成
  - [CLS]トークンに対応するHidden vectorを文対表現として全結合層に入力、二値分類

### Fine-tuning tasks
MSCOCOとFlickr30kデータセットで実験を行った
学習に使用した損失関数
- Binary classification Loss
  - textとimageが一致しているかのやつ
- Multi-class Classification Loss
  -  正しい組み合わせと間違った組み合わせのマージンを最大化するのによく使われるらしい
  - 正しいl(v, w)のラベルを予測する
- Triplet Loss
  - 正しい組み合わせに近い間違った組み合わせとのマージンを最大化するのに使用

### 実験結果
- テキストが与えられた時の画像検索
- 画像が与えられた時のテキスト検索
- 評価指標：Recall@k
  - 検索結果上位k個のRecall
### Pre-trained Modelの評価
![image](https://user-images.githubusercontent.com/38539915/79510165-ce5ebc00-8077-11ea-9bdf-5c9a59cf6508.png)
- Zero-shotなので、fine-tune無しでpre-trainだけした時の結果
- UNITERよりは悪いが、MSCOCOではSoTA

### Fine-tuned Modelの評価
![image](https://user-images.githubusercontent.com/38539915/79510186-d880ba80-8077-11ea-860c-2fc8070b0548.png)
- fine-tuneした場合
- SoTA

### Ablation
![image](https://user-images.githubusercontent.com/38539915/79510211-e33b4f80-8077-11ea-92b3-d912cdff4e3a.png)
- Multi-stage Pretrainingが良いという話
- LAIT単体だとあまりよくない
- しかし、LAITでpre-trainすることで、CC+SBUだけ使うより2~6ポイント改善
